moveit_setup_assistant/include/moveit/setup_assistant/tools/moveit_config_data.h

```diff
@@ -157,7 +157,11 @@ class MoveItConfigData
   bool urdf_from_xacro_;
 
   /// URDF robot model
+#if __cplusplus <= 199711L
   boost::shared_ptr<urdf::Model> urdf_model_;
+#else
+  std::shared_ptr<urdf::Model> urdf_model_;
+#endif
```

Comment:
I don't think `#if __cplusplus <= 199711L` is the correct thing to select the `shared_ptr` type on though. It is possible to compile moveit with C++11 support with an older `urdfdom_headers` that still uses `boost::shared_ptr`.

You could use something like this (maybe move it to moveit_core to avoid duplication):
https://github.com/ros-planning/moveit/blob/d001746257005e7bde01f1bed581ab78a763d3f0/moveit_planners/ompl/ompl_interface/include/moveit/ompl_interface/detail/same_shared_ptr.hpp